test/unit/event.js

```diff
@@ -5,20 +5,6 @@ module( "event", {
 	teardown: moduleTeardown
 });
 
-test("null or undefined handler", function() {
-	expect(2);
-	// Supports Fixes bug #7229
-	try {
-		jQuery("#firstp").on( "click", null );
-		ok(true, "Passing a null handler will not throw an exception");
-	} catch ( e ) {}
-
-	try {
-		jQuery("#firstp").on( "click", undefined );
-		ok(true, "Passing an undefined handler will not throw an exception");
-	} catch ( e ) {}
-});
-
```

Comment:
@markelog The behavior is defined because we said so. We don't need to document that invalid input will throw. That is assumed. What I'm advocating for here is that we add tests to ensure the behavior that we want to happen actually happens. It prevents us from changing the behavior in the future unless we do it intentionally.